src/renderer/components/ContextMenu.tsx

```diff
@@ -197,15 +247,20 @@ export function ContextMenu(props: {
     }
 
     const onOutsideClick = (ev: MouseEvent | TouchEvent) => {
-      const parent = document.querySelector('div.dc-context-menu')
-      if (!parent) {
-        return
-      }
-
-      let isOnMenu = ev.target === parent
-      for (const child of parent.children) {
-        if (ev.target === child) {
-          isOnMenu = true
+      let isOnMenu = false
+
+      for (const menuEl of menuLevelEls.current) {
+        isOnMenu = ev.target === menuEl
+        if (!isOnMenu) {
+          for (const child of menuEl.children) {
+            if (ev.target === child) {
+              isOnMenu = true
+              break
+            }
+          }
+        }
```

Comment:
This is a fairly high degree of nesting which makes the code hard to read, do you mind refactoring it if possible?

Reply:
How about using `map` and/or `filter`? I think they are also faster?